### ISSUE TITLE
Create the parameters for train_model and the base training loop for train_model

### ISSUE THREAD

**Issue by Jacob-Meert** [psu-nittanyaiadvance/26-S-Lockheed-2#27]
### Description of the Task:

Fill out the parameter section of the train_model function. This list of parameters should be all encompassing given our current understanding of the model training (but of course we can change them later). Write a loop that will run once for each epoch of training. The beginning of this loop should also set the model to training mode. The loop should also maintain a progress bar for monitoring the training progress in the console (with tqdm).

### Acceptance Criteria:

-  All of the immediately obvious parameters for the train_model function are present
        - Where appropriate add parameter defaults
- The training loop runs for the number of epochs passed to the function
- When the training loop is begun, the model is placed in training mode
- There is a detailed progress bar that will update the trainer on model progress
- Code should be entirely understood (including theory behind decisions)
